src/flint/FlintTypes.jl

```diff
@@ -1771,7 +1774,9 @@ mutable struct FqFiniteField <: FinField
       end
    end
 
-   function FqFiniteField(f::fmpz_mod_poly, s::Symbol, cached::Bool = true)
+   function FqFiniteField(f::fmpz_mod_poly, s::Symbol, cached::Bool = true; check::Bool = true)
+      check && !isprime(modulus(f)) &&
```

Comment:
We think that `isprime` is too expensive. Can you change this to `isprobable_prime` (or whatever the name is). Once this is done, we can merge this.

Reply:
Ok will do. Note that `isprime` is used in a couple other places in this file (`FlintPadicField` and `FlintQadicField`), would you like this to be changed too? (in a separate PR).